### PR TITLE
Added more detail to guide

### PR DESCRIPTION
**Pull request by andrewdes**
Implemented feedback from https://github.com/OpenLiberty/draft-guide-jpa-intro/issues/32 and https://github.com/OpenLiberty/draft-guide-jpa-intro/issues/29

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -16,7 +16,7 @@
 :page-permalink: /guides/{projectid}
 :common-includes: https://raw.githubusercontent.com/OpenLiberty/guides-common/master
 :source-highlighter: prettify
-= Performing CRUD operations using Java Persistence API
+= Persisting and accessing data using JPA
```
Comment (jgrassel) [inline README.adoc]: I think I'd still say **Java Persistence API (JPA)** here.

File: README.adoc [hunk 2]
```diff
@@ -29,20 +29,31 @@ Learn how to map Java objects to database tables and perform CRUD operations usi
 
 == What you'll learn
 
-You will learn how to use the Java Persistence API (JPA) to map Java objects to database tables.
-JPA is a standard for object relational mapping (ORM) and eliminates the burden of having to write
-SQL code when performing database operations.
+You will learn how to use the Java Persistence API (JPA) to map Java objects to database tables and
+perform create, read, update and delete (CRUD) operations. 
+
+JPA is a Java specification for persisting Java objects to a relational database. The JPA specification
```
Comment (jgrassel) [inline README.adoc]: I dunno, maybe describing it as a technology that represents RDBMS table data as Plain Old Java Objects (POJO) might be better?

File: README.adoc [hunk 3]
```diff
@@ -29,20 +29,31 @@ Learn how to map Java objects to database tables and perform CRUD operations usi
...
+is implemented by several different persistence providers. In this guide you will use EclipseLink as
+the persistence provider.
```
Comment (jgrassel) [inline README.adoc]: Maybe clarify that Eclipselink is the JPA Provider shipped with Liberty, just so the user knows that it isn't a randomly chosen JPA Persistence Provider Implementation here.

File: README.adoc [hunk 4]
```diff
@@ -29,20 +29,31 @@ Learn how to map Java objects to database tables and perform CRUD operations usi
...
+
+JPA simplifies object relational mapping (ORM) by using annotations or XML to map Java objects
```
Comment (jgrassel) [inline README.adoc]: Make that "annotations and/or XML", they're not mutually exclusive.  You can use annotations and XML, they aggregate together (unless you select XML Metadata Complete), and XML overrides annotation values.

File: README.adoc [hunk 5]
```diff
@@ -136,8 +153,8 @@ include::finish/src/main/resources/META-INF/persistence.xml[tags=**;]
 
 The persistence unit is defined by the `<persistence-unit />` XML element. The `name` attribute is 
 required and is used to identify the persistent unit. The `transaction-type=”JTA”` element specifies 
-to use Java Transaction API (JTA) transactions. In a Java EE environment, the default is JTA if this 
-element is not specified.
+to use Java Transaction API (JTA) to manage the EntityManager's transactions. Since we are using
```
Comment (jgrassel) [inline README.adoc]: Maybe something like:

> The 'transaction-type="JTA"' element specifies that the persistence context will enlist with Java Transaction API (JTA) global transactions.

Comment (jgrassel) [reply README.adoc]: I say that because JPA will not do anything to control global transactions -- all it will do is either mark it for rollback, or respond to transaction callbacks (such as post-completion, etc.)  The only transactions that the JPA provider will directly control are the resource local transactions, those managed by EntityTransaction.

All that is just for your info, you don't need to mention that in the guide here.

File: README.adoc [hunk 6]
```diff
@@ -172,8 +189,13 @@ and whether to create the database if it does not already exist.
 
 == Performing CRUD operations using JPA
 
-CRUD is an abbreviation that defines the four basic functionalities that can be performed on a 
-database: create, read, update, and delete.
+Every entity object has a lifecycle that consists of four states: new, managed, detached and removed.
```
Comment (jgrassel) [inline README.adoc]: Removed really isn't a "state", when you call em.remove(), that entity is marked for removal. It's still managed until you punt it with em.clear() or em.detach() or complete the transaction.

File: README.adoc [hunk 7]
```diff
@@ -184,33 +206,39 @@ include::finish/src/main/java/io/openliberty/guides/jpaguide/dao/EventDao.java[t
...
 In order to use the entity manager at runtime, it must be injected into our EJB component through the
-`@PersistenceContext` annotation. Since the entity manager is container-managed, the EJB container
-will ensure that the methods are executed within an active transaction which is why it is not necessary
-to explicitly begin, commit, or rollback transactions. At the end of an EJB operation, the persistence 
-context will be notified that the transaction is committing and the persistence provider, EclipseLink,
-will flush the changes to the database.
+`@PersistenceContext` annotation. Since the stateless session bean is configured to manage the global
+transactions, the EJB container will ensure that the required methods are executed within the boundaries
+of an active global transaction, which is why it is not necessary to explicitly begin, commit or rollback
+transactions. At the end of a session bean method invocation, the persistence context will be notified
```
Comment (jgrassel) [inline README.adoc]: That's effectively true for the guide's application that I've seen so far.  But generally speaking, the business method that began the transaction in its preInvoke phase is the same business method that will commit/rollback the transaction in its postInvoke phase.

Things get more complicated when you have session beans business methods calling the business methods of other session beans, because you have the elements of transaction propagation (or transaction suspension) and persistence context propagation.

File: README.adoc [hunk 8]
```diff
@@ -184,33 +206,39 @@ include::finish/src/main/java/io/openliberty/guides/jpaguide/dao/EventDao.java[t
...
-method must be an instance of an entity class, otherwise an `IllegalArgumentException` will be thrown.  
+calling the `persist()` method on an EntityManager instance. The event objects state will be 
+changed from new to managed and changes to it are now tracked by the EntityManager. 
 
 * The `readEvent()` method will return an instance of the `Event` entity class with the specified primary 
 key by calling the `find()` method on an `EntityManager` instance. The `find()` method accepts both the entity 
-class and the primary key as arguments. If the event instance is not found, `Null` will be returned. Similarily,
-the `readAllEvents()` method returns a list of instances of the `Event` entity class using a query
-specified in the `@NamedQuery` annotation on the `Event` class. 
+class and the primary key as arguments. If the state of the entity being searched for is managed, the event
```
Comment (jgrassel) [inline README.adoc]: Depends.  A simple em.find() (which is a single-entity single-identity fetch request) will certainly check the persistence context first before issuing SQL statements to the database, but a query will still issue SQL statements to the database.

File: README.adoc [hunk 9]
```diff
@@ -184,33 +206,39 @@ include::finish/src/main/java/io/openliberty/guides/jpaguide/dao/EventDao.java[t
...
+
+* The `readAllEvents()` method demonstrates an alternate way to retrieve event objects from the database.
+This method returns a list of instances of the `Event` entity class using a query specified in the `@NamedQuery` 
+annotation on the `Event` class. 
 
-* The `updateEvent()` method updates an instance of the `Event` entity class in the database. The 
-`EntityManager` automatically tracks entity objects in the persistence context for changes and synchronizes
-them with the database. The `merge()` method will copy the entity supplied as an argument into a new 
-`Event` instance that will be added to the persistence context so that its changes can be tracked.
+* The `updateEvent() method creates a managed instance of a detached entity object. The EntityManager
```
Comment (jgrassel) [inline README.adoc]: Might want to explain why the entity is detached, and explain that the merge operation not only returns a managed instance of the merge() argument but also includes changes made to the detached entity's loaded fields.
